modules/core/src/test/java/org/apache/ignite/internal/IgniteNodeValidationFailedEventTest.java

```diff
@@ -0,0 +1,89 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal;
+
+import java.util.UUID;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.events.Event;
+import org.apache.ignite.events.NodeValidationFailedEvent;
+import org.apache.ignite.internal.processors.security.impl.TestSecurityPluginProvider;
+import org.apache.ignite.spi.IgniteSpiException;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+import static org.apache.ignite.events.EventType.EVT_NODE_VALIDATION_FAILED;
+import static org.apache.ignite.internal.IgniteNodeAttributes.ATTR_NODE_CONSISTENT_ID;
+import static org.apache.ignite.plugin.security.SecurityPermissionSetBuilder.ALLOW_ALL;
+import static org.apache.ignite.testframework.GridTestUtils.assertThrowsWithCause;
+
+/**
+ * Tests joining node validation failed event.
+ */
+public class IgniteNodeValidationFailedEventTest extends GridCommonAbstractTest {
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        return super.getConfiguration(igniteInstanceName)
+            .setIncludeEventTypes(EVT_NODE_VALIDATION_FAILED)
+            .setConsistentId(UUID.randomUUID());
+    }
+
+    /**
+     * @throws Exception If failed.
+     */
+    @Test
+    @SuppressWarnings("ThrowableNotThrown")
+    public void testNodeValidationFailedEvent() throws Exception {
+        startGrid(0);
+
+        CountDownLatch evtLatch = new CountDownLatch(1);
+
+        AtomicReference<Event> listenedEvt = new AtomicReference<>();
+
+        grid(0).events().localListen(evt -> {
+            assertTrue(listenedEvt.compareAndSet(null, evt));
+
+            evtLatch.countDown();
+
+            return true;
+        }, EVT_NODE_VALIDATION_FAILED);
+
+        startGrid(1);
+
+        IgniteConfiguration invalidCfg = getConfiguration(getTestIgniteInstanceName(2))
+            .setPluginProviders(new TestSecurityPluginProvider("login", "", ALLOW_ALL, false));
+
+        assertThrowsWithCause(() -> startGrid(optimize(invalidCfg)), IgniteSpiException.class);
+
+        evtLatch.await();
+
+        assertTrue(listenedEvt.get() instanceof NodeValidationFailedEvent);
+
+        NodeValidationFailedEvent validationEvt = (NodeValidationFailedEvent) listenedEvt.get();
```

Comment:
double get should be replaced with single

Reply:
Thanks. Done.